src/components/button/button.tsx

```diff
@@ -204,6 +204,10 @@ export class Button implements LabelableComponent, InteractiveComponent, FormOwn
       <Tag
         aria-label={getLabelText(this)}
         class={{
+          [CSS.buttonNoBorderPadding]: !this.iconStart && !this.iconEnd,
```

Comment:
Maybe add a const for `!this.iconStart && !this.iconEnd`?

Like 

```ts
const noStartEndIcons = !this.iconStart && !this.iconEnd;
```

```ts
[CSS.buttonNoBorderPadding]: noStartEndIcons,
[CSS.buttonNoBorderPaddingShrunk]: !noStartEndIcons,
[CSS.buttonPadding]: noStartEndIcons,
[CSS.buttonPaddingShrunk]: !noStartEndIcons,
```

Looking at the above, it seems that two of those classes are the same right? Could it be reduced to...


```ts
[CSS.noStartEndIcons]: noStartEndIcons,
[CSS.hasStartEndIcons]: !noStartEndIcons,
```